src/rc.js

```diff
@@ -5,17 +5,15 @@ import parse from './lockfile/parse.js';
 import * as rcUtil from './util/rc.js';
 
 // Keys that will get resolved relative to the path of the rc file they belong to
-const PATH_KEYS = ['cache-folder', 'global-folder', 'modules-folder'];
+const PATH_KEYS = ['cache-folder', 'global-folder', 'modules-folder', 'cwd'];
 
-let rcConfCache;
-let rcArgsCache;
-
-const buildRcConf = () =>
-  rcUtil.findRc('yarn', (fileText, filePath) => {
+// given a cwd, load all .yarnrc files relative to it
+function getRcConfigForCwd(cwd: string): {[key: string]: string} {
+  return rcUtil.findRc('yarn', cwd, (fileText, filePath) => {
     const {object: values} = parse(fileText, 'yarnrc');
-    const keys = Object.keys(values);
 
-    for (const key of keys) {
+    // some keys reference directories so keep their relativity
+    for (const key in values) {
```

Comment:
I'd say the problem is more readability than anything. It's really unusal to see `for ... in` loops nowadays, and it's not obvious when reading the code whether it's intended or not. Flow helps a bit, but not much.

That being said, `for ... in` loops are used here and there in the source code so if you want to use it there I'm fine with it. If we eventually choose to stop using them it will be in another PR.